### DISCUSSION TITLE
How to fix NextJs proxying with rewrites?

### DISCUSSION THREAD

**Discussion by koval96** [vercel/next.js#41388] [Help] [answered]
I need to proxy api requests to another host from next dev server. I'm using rewrites in my next config, however proxying not working.

My `next.config.js`:
```
const withReactSvg = require('next-react-svg')
const path = require('path')

module.exports = () => {
  const rewrites = () => {
    return [
      {
        source: '/api/v1/:path*',
        destination: 'api_host/api/v1/:path*' // Proxy to Backend
      }
    ]
  }
  return {
    rewrites,
    include: path.resolve(__dirname, 'src/assets/svg'),
    webpack(config, options) {
      config.module.rules.push({
        test: /\.svg$/,
        use: ["@svgr/webpack"]
      });
      return config
    },
  }
}

const nextConfig = {
  reactStrictMode: true,
}

module.exports = nextConfig
```

When trying to request api with `axios("/api/v1/getUser"`), I get the following error: Request failed with status code 404, because proxying not working and the request goes to `localhost:3000/api/v1/getUser`

**Comment by samithaf** [comment, marked as answer]
If the api server is running in the same port, can you update the config as follows?

`destination: '/api_host/api/v1/:path*' `

If your api server is an external source then you will need to define the full URL. https://nextjs.org/docs/api-reference/next.config.js/rewrites#rewriting-to-an-external-url
